tasks/eva_2357/populate_ids.py

```diff
@@ -0,0 +1,196 @@
+        return by_refseq[contig]['genbank'], False
+    raise Exception(f"Could not find synonym for contig {contig}")
+
+
+def get_SHA1(variant_id):
+    """Calculate the SHA1 digest from the seq, study, contig, start, ref, and alt attributes of the variant"""
+    return hashlib.sha1(variant_id.encode()).hexdigest().upper()
+
+
+def get_ids(assembly, contig_synonym_dictionaries, variant_query_result, hash_to_variant_id):
+    """
+    Get a dictionary with the submitted variant hash as key and the variant id (chr_start_ref_alt) from the variant
+    warehouse as value
+    """
+    for file in variant_query_result['files']:
+        study = file['sid']
+        start = variant_query_result['start']
+        ref = variant_query_result['ref']
+        alt = variant_query_result['alt']
+        id_variant_warehouse = variant_query_result['_id']
+        try:
+            genbank_chr, was_already_genbank = get_genbank(contig_synonym_dictionaries, variant_query_result['chr'])
+        except Exception as e:
+            logging.info(f"{e} in variant {variant_query_result['chr']}_{start}_{ref}_{alt}")
+            continue
+        sve_hash = get_SHA1(f"{assembly}_{study}_{genbank_chr}_{start}_{ref}_{alt}")
+        hash_to_variant_id[sve_hash] = id_variant_warehouse
+    return hash_to_variant_id
+
+
+def get_db_name_and_assembly_accession(databases):
+    db_assembly = {}
+    if databases:
+        with open(databases, 'r') as db_file:
+            for line in db_file:
+                db, assembly, asm_report = line.split(',')
+                db_assembly[db] = {"assembly": assembly, "asm_report": asm_report}
+        return db_assembly
+
+
+def populate_ids(private_config_xml_file, databases, profile='production', mongo_accession_db='eva_accession_sharded'):
+    db_assembly = get_db_name_and_assembly_accession(databases)
+    for db_name, info in db_assembly.items():
+        assembly = info['assembly']
+        asm_report = info['asm_report']
+        logging.info(f"Processing database {db_name} (assembly {assembly})")
+
+        contig_synonym_dictionaries = load_synonyms_for_assembly(assembly, asm_report)
+
+        with pymongo.MongoClient(get_mongo_uri_for_eva_profile(profile, private_config_xml_file)) as mongo_handle:
+            variants_collection = mongo_handle[db_name]["variants_2_0"]
+            logging.info(f"Querying variants from variant warehouse, database {db_name}")
+            variants_cursor = get_variants_from_variant_warehouse(variants_collection)
+            hash_to_variant_id = {}
+            update_statements = []
+            try:
+                for variant_query_result in variants_cursor:
+                    # One variant in the variant warehouse can include more than one study, which means there can be
+                    # more than one submitted variant per variant warehouse variant
+                    get_ids(assembly, contig_synonym_dictionaries, variant_query_result, hash_to_variant_id)
```

Comment:
To avoid implicit mutation of the passed dict, I suggest 
```suggestion
                   hash_to_variant_id = get_ids(assembly, contig_synonym_dictionaries, variant_query_result, hash_to_variant_id)
```

Reply:
I would go one step further and not mutate the dict at all in the function
```suggestion
                    hash_to_variant_id.update(get_ids(assembly, contig_synonym_dictionaries, variant_query_result))
```